### PR TITLE
Jon/fix/fio-tx-saved-data

### PR DESCRIPTION
**Pull request by Jon-edge**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [x] Yes
- [ ] No

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208058760262042

### CODE REVIEW

**Review by peachbits — CHANGES_REQUESTED** [review]
This may resolve the unstaking bug but I think the refactor introduced new bugs for other transaction types.

File: src/fio/FioEngine.ts [hunk 1]
```diff
@@ -2046,6 +2036,20 @@ export const parseAction = ({
     meta: {}
   }
 
+  // Check if this is an unstake, and update the staking/locked status later
+  // with this info.
+  const isUnstakeRewardTx =
+    otherParams.data != null && otherParams.data.memo === STAKING_REWARD_MEMO
```
[the remark this answers is not in the document]
Comment (Jon-edge) [reply src/fio/FioEngine.ts:2042]: https://gist.githubusercontent.com/Jon-edge/f2a51410ab1b9848594b4b7387e760a6/raw/c5a8ff74c638a6d3d67a5683ea93ad1169dd7575/FIO%20stake---unstake%20bug.json

File: src/fio/FioEngine.ts [hunk 2]
```diff
@@ -2139,11 +2147,9 @@ export const parseAction = ({
           networkFee = '0'
         } else {
           networkFee = isRecipient ? `-${fioAmount}` : fioAmount
+          networkFee = '0'
```
Comment (peachbits) [inline src/fio/FioEngine.ts:2156]: This looks like a typo

File: src/fio/FioEngine.ts [hunk 3]
```diff
@@ -2139,11 +2147,9 @@ export const parseAction = ({
...
         }
 
-        nativeAmount = data.to === actor ? `${fioAmount}` : `-${fioAmount}`
```
Comment (peachbits) [inline src/fio/FioEngine.ts]: I think these two lines should be removed from the previous commit

**Review by peachbits — CHANGES_REQUESTED** [review]
Much better! Only two minor issues then it looks good to go

File: src/fio/FioEngine.ts (hunk 2 above)
Comment (Jon-edge) [reply src/fio/FioEngine.ts:2156]: Yeah bad merge conflict resolution, good catch

**Review by peachbits — APPROVED** [review]

**Pull request merged**